### PR TITLE
chore(numericSelector): initial import

### LINKED ISSUE #563: API proposal: numeric selector
The idea is to create a new widget that allows you to build the Airbnb-like "Guests" select:

I was thinking about something like:

```js
search.addWidget(
  instantsearch.widgets.numericSelector({
    container: '#guests',
    attributeName: 'person_capacity',
    operator: '>=',
    options: [
      { label: '1 guest', value: 1 },
      { label: '2 guests', value: 2 },
      { label: '3 guests', value: 3 },
      { label: '4 guests', value: 4 },
      { label: '5 guests', value: 5 },
      { label: '6 guests', value: 6 }
    ]
  })
);
```

### PR DESCRIPTION
**Pull request by redox**
A new widget to build dropdown menu to refine on a numeric attribute.

(This one is also mandatory for our examples).

Fix #563

### CODE REVIEW

File: docs/documentation.md [hunk 1]
```diff
@@ -774,6 +774,47 @@ instantsearch.widgets.priceRanges(options);
 
 <div id="price-ranges" class="widget-container"></div>
 
+#### numericSelector
+
+<div class="codebox-combo">
+
+<img class="widget-icon pull-left" src="../img/icon-widget-slider.svg">
```
Comment (pixelastic) [inline docs/documentation.md]: We'll need a new icon

File: widgets/numeric-selector/__tests__/numeric-selector-test.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    numericSelector.__Rewire__('autoHideContainerHOC', autoHideContainer);
+
+    container = document.createElement('div');
+    options = [
+      {value: 1, label: 'first'},
+      {value: 2, label: 'second'}
+    ];
+    cssClasses = {
+      root: 'custom-root',
+      item: 'custom-item'
```
Comment (pixelastic) [inline widgets/numeric-selector/__tests__/numeric-selector-test.js]: The doc says it's called `select`

File: docs/documentation.md [hunk 3]
```diff
@@ -774,6 +774,47 @@ instantsearch.widgets.priceRanges(options);
...
+    container: '#popularity-selector',
+    attributeName: 'popularity',
+    operator: '<=',
+    options: [
+      { label: 'Top 1', value: 1 },
+      { label: 'Top 10', value: 10 },
+      { label: 'Top 100', value: 100 }
+    ],
+    cssClasses: {
+      select: ''
```
Comment (pixelastic) [inline docs/documentation.md]: The code says it's `root` and `item`

File: widgets/numeric-selector/numeric-selector.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+          options={options}
+          setValue={this._refine.bind(this, helper)}
+          shouldAutoHideContainer={hasNoResults}
+        />,
+        containerNode
+      );
+    },
+
+    _refine: function(helper, value) {
+      if (value !== undefined) {
```
Comment (pixelastic) [inline widgets/numeric-selector/numeric-selector.js]: Shouldn't we `clearRefinement` when `value === undefined`?

Comment (redox) [inline widgets/numeric-selector/numeric-selector.js]: Good point! Definitely.

File: docs/documentation.md (hunk 3 above)
Comment (redox) [inline docs/documentation.md]: Correct, I've copy/pasted the `hitsPerPageSelector` -> I'll update it as well.

**Comment by redox** [comment]
I'm good @pixelastic

**Comment by redox** [comment]
rebased

File: widgets/numeric-selector/numeric-selector.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      helper.search();
+    },
+
+    _getRefinedValue: function(helper) {
+      const refinements = helper.getRefinements(attributeName);
+      let res;
+      refinements.forEach(v => {
+        if (v.operator === operator) {
+          res = v.value[0];
+          return false;
```
Comment (bobylito) [inline widgets/numeric-selector/numeric-selector.js]: What is `return false` supposed to do?

Comment (redox) [inline widgets/numeric-selector/numeric-selector.js]: Arg that was an artefact of the `$.each` function to stop the enumeration.

Comment (vvo) [inline widgets/numeric-selector/numeric-selector.js]: You can use lodash.find (Array.find being ES6). We already use it in other places

**Comment by redox** [comment]
@bobylito fixed!

File: docs/documentation.md [hunk 6]
```diff
@@ -753,13 +754,54 @@ instantsearch.widgets.priceRanges(options);
...
+<div id="popularity-selector" class="widget-container"></div>
+
 #### clearAll
 
 <div class="codebox-combo">
 
 <img class="widget-icon pull-left" src="../img/icon-widget-clearall.svg">
-This filtering widget lets the user choose between ranges of price. Those ranges are dynamically computed based on the returned results.
-{:.description}
+This widget clears all the refinements that are currently applied.
```
Comment (redox) [inline docs/documentation.md]: @Jerskouille I've update the documentation of the `clearAll` widget as well.

File: widgets/numeric-selector/numeric-selector.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      helper.clearRefinements(attributeName);
+      if (value !== undefined) {
+        helper.addNumericRefinement(attributeName, operator, value);
+      }
+      helper.search();
+    },
+
+    _getRefinedValue: function(helper) {
+      const refinements = helper.getRefinements(attributeName);
+      return refinements.find(v => {
```
Comment (vvo) [inline widgets/numeric-selector/numeric-selector.js]: Cannot use refinements.find, not available in most browsers (this is ES6). We will fail the build soon on those ones, by running unit test in real browsers

Comment (redox) [inline widgets/numeric-selector/numeric-selector.js]: Arg OK, I thought babel would transpile it to something <ES2015 compliant :/

**Comment by redox** [comment]
Done @vvo @bobylito

**Comment by vvo** [comment]
@redox just so you know, babel only does syntax transformation to es5, not actual shims (a bit tricky). We ca automatically include a shim but we already have lodash everywhere

File: widgets/numeric-selector/numeric-selector.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+      helper.clearRefinements(attributeName);
+      if (value !== undefined) {
+        helper.addNumericRefinement(attributeName, operator, value);
+      }
+      helper.search();
+    },
+
+    _getRefinedValue: function(helper) {
+      const refinements = helper.getRefinements(attributeName);
+      return find(refinements, {operator: operator});
```
Comment (bobylito) [inline widgets/numeric-selector/numeric-selector.js]: shortcut : `{operator}` for `{operator: operator}`

**Comment by vvo** [comment]
LGTM

**Comment by bobylito** [comment]
LGTM :+1:

**Pull request merged**